### PR TITLE
Feature/8 create base classes to follow architecture guidelines

### LINKED ISSUE #8: Create base classes to follow architecture guidelines
The project needs to implement base classes to provide guidelines for future concrete implementations, in case of the addition of any dataset, it should follow the usage of base classes to download and upload data.

mdsets model

### PR DESCRIPTION
**Pull request by alejocas**
Closes #8

### CODE REVIEW

File: mdsets/core/base/BaseConnector.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+from abc import ABC, abstractmethod
+
+
+class BaseConnector(ABC):
+    """
+    Class intended to define the downloading and uploading mechanisms that
```
Comment (sebasarango1180) [inline mdsets/core/base/BaseConnector.py:6]: All lines of code must be compliant with Black formatting rule: `line-length=88`.
Fixed already ✔️

File: mdsets/core/base/BaseDestination.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
+from abc import ABC, abstractmethod
+
+
+class BaseDestination(ABC):
+    """
+    Base class to define which methods are to be implemented by every uploader
```
Comment (sebasarango1180) [inline mdsets/core/base/BaseDestination.py:6]: Same as above. The command is, from the project root:
```bash
black .

File: mdsets/core/connector/ExampleConnector.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
+from mdsets.core.base.BaseConnector import BaseConnector
+
+
+class ExampleConnector(BaseConnector):
```
Comment (sebasarango1180) [inline mdsets/core/connector/ExampleConnector.py:4]: Examples for abstracts?
image

**Review by sebasarango1180 — APPROVED** [review]
Nice work
LGTM 🚀

File: mdsets/core/connector/ExampleConnector.py (hunk 3 above)
Comment (alejocas) [reply mdsets/core/connector/ExampleConnector.py:4]: Feels weird but necessary

**Pull request merged**